core/src/commonMain/kotlin/com/xebia/functional/xef/vectorstores/LocalVectorStore.kt

```diff
@@ -31,15 +31,27 @@ private constructor(private val embeddings: Embeddings, private val state: Atomi
 
   override suspend fun addMemories(memories: List<Memory>) {
     state.update { prevState ->
+      val allNewMemories = memories.groupBy { it.conversationId }
+
+      val prevConversationIdsInNewMemories = prevState.orderedMemories.map { it.key }.intersect(allNewMemories.keys)
+
+      val prevMemoriesWithNewMessages = prevState.orderedMemories.map { (conversationId, memories) ->
+        conversationId to (memories + allNewMemories[conversationId].orEmpty())
+      }.toMap()
+
+      val allNewMemoriesWithoutPrevConversationId = allNewMemories.filterKeys {
+        !prevConversationIdsInNewMemories.contains(it)
+      }
+
       prevState.copy(
-        orderedMemories = prevState.orderedMemories + memories.groupBy { it.conversationId }
+        orderedMemories = prevMemoriesWithNewMessages + allNewMemoriesWithoutPrevConversationId
       )
     }
   }
 
   override suspend fun memories(conversationId: ConversationId, limit: Int): List<Memory> {
     val memories = state.get().orderedMemories[conversationId]
-    return memories?.take(limit).orEmpty()
+    return memories?.takeLast(limit).orEmpty()
```

Comment:
@javipacheco and I have been talking, and I agree with this change to provide `memories` in the order they were created (that is, from oldest to newest), and getting (up to) the last `limit` ones. Now I'm wondering if we should consider this to happen for all possible implementations of `VectorStore`, providing users with the postcondition that `memories` always returns values ordered from oldest to newest and we return up to `limit` of them. Does that make sense, @raulraja? Is the intended usage/implementation different? In that case, we should revisit the other `VectorStore` implementations to make sure that that postconditions are always met. I propose it, and working on that on a different PR

Reply:
I agree @victorcrrd, if this is the way to do that, we should review all of the implementations of `VectorStore` and we must create tests for every implementation

Reply:
We just need to make sure we are converting the memories into messages in the appropriate order as the chat completion endpoint expects them. See open ai api docs for details of what the conversation looks like in terms of order

Reply:
The [OpenAI documentation](https://platform.openai.com/docs/api-reference/chat/create) does not explicitly state that the order of messages has to be in ascending order, although [all examples](https://github.com/openai/openai-cookbook/blob/main/examples/How_to_format_inputs_to_ChatGPT_models.ipynb) show that they should be sent in ascending order.

I have been doing some tests in postman and when I change the order of messages, I receive different results. 

If we want to change all of the implementations of `VectorStore` we need to change the following:

- `CombinedVectorStore`
- `Lucene`
- `PGVectorStore`: maybe we need `test-containers` for creating tests in this store

I think that we should address this in separated PRs

Reply:
Let's do this in this PR. Let me know if there is anything I can do to help